enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/storecopy/LocalDatabase.java

```diff
@@ -41,53 +41,60 @@
     private final CopiedStoreRecovery copiedStoreRecovery;
     private final StoreFiles storeFiles;
     private final DataSourceManager dataSourceManager;
-    private final Supplier<TransactionIdStore> transactionIdStoreSupplier;
+    private final Supplier<TransactionIdStore> onlineTxIdStoreSupplier;
+    private final Supplier<TransactionIdStore> offlineTxIdStoreSupplier;
     private final Supplier<DatabaseHealth> databaseHealthSupplier;
+    private final Log log;
 
     private volatile StoreId storeId;
     private volatile DatabaseHealth databaseHealth;
-    private final Log log;
+    private boolean started;
 
     public LocalDatabase( File storeDir, CopiedStoreRecovery copiedStoreRecovery, StoreFiles storeFiles,
-                          DataSourceManager dataSourceManager, Supplier<TransactionIdStore> transactionIdStoreSupplier,
-                          Supplier<DatabaseHealth> databaseHealthSupplier, LogProvider logProvider )
+            DataSourceManager dataSourceManager, Supplier<TransactionIdStore> onlineTxIdStoreSupplier,
+            Supplier<TransactionIdStore> offlineTxIdStoreSupplier, Supplier<DatabaseHealth> databaseHealthSupplier,
+            LogProvider logProvider )
     {
         this.storeDir = storeDir;
         this.copiedStoreRecovery = copiedStoreRecovery;
         this.storeFiles = storeFiles;
         this.dataSourceManager = dataSourceManager;
-        this.transactionIdStoreSupplier = transactionIdStoreSupplier;
+        this.onlineTxIdStoreSupplier = onlineTxIdStoreSupplier;
+        this.offlineTxIdStoreSupplier = offlineTxIdStoreSupplier;
         this.databaseHealthSupplier = databaseHealthSupplier;
-        log = logProvider.getLog( getClass() );
+
         this.storeId = StoreId.DEFAULT;
+        this.log = logProvider.getLog( getClass() );
     }
 
     @Override
-    public void init() throws Throwable
+    public synchronized void init() throws Throwable
```

Comment:
Just for symmetry of all the lifecycle stages, where at least start/stop now, because of started, shouldn't be running concurrently.